XCode project generation error with CMake 3.19.3

When trying to generate XCode project with CMake 3.19.3, the following error occurs:

```
CMake Error in source/CMakeLists.txt:
  The custom command generating

    /Users/<user>/git/SPIRV-Tools/build/build-version.inc

  is attached to multiple targets:

    SPIRV-Tools-shared
    spirv-tools-build-version
    SPIRV-Tools-static

  but none of these is a common dependency of the other(s).  This is not
  allowed by the Xcode "new build system".
```

This error only started happening in CMake 3.19 and there was no such problem in 3.18.
It is possible to workaround this issue using  `-T buildsystem=1` CMake argument, but maybe there should be a better solution.

SPIRV-Tools version: https://github.com/KhronosGroup/SPIRV-Tools/commit/cec658c11603a5411a38f0f86d0ecc04250df4b7
SPIRV-Headers version: https://github.com/KhronosGroup/SPIRV-Headers/commit/f027d53ded7e230e008d37c8b47ede7cd308e19d

CMake command: 

```
mkdir build
cd build
cmake -G "Xcode" ..
```

--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -359,7 +359,7 @@ function(spirv_tools_default_target_options target)
   )
   set_property(TARGET ${target} PROPERTY FOLDER "SPIRV-Tools libraries")
   spvtools_check_symbol_exports(${target})
-  add_dependencies(${target} core_tables enum_string_mapping extinst_tables)
+  add_dependencies(${target} spirv-tools-build-version core_tables enum_string_mapping extinst_tables)
 endfunction()
 
 # Always build ${SPIRV_TOOLS}-shared. This is expected distro packages, and

